src/server.c

```diff
@@ -502,16 +517,17 @@ dictType kvstoreExpiresDictType = {
     kvstoreDictMetadataSize,
 };
 
-/* Command table. sds string -> command struct pointer. */
-dictType commandTableDictType = {
-    dictSdsCaseHash,            /* hash function */
-    NULL,                       /* key dup */
-    dictSdsKeyCaseCompare,      /* key compare */
-    dictSdsDestructor,          /* key destructor */
-    NULL,                       /* val destructor */
-    NULL,                       /* allow to expand */
-    .no_incremental_rehash = 1, /* no incremental rehash as the command table may be accessed from IO threads. */
-};
+/* Command set, hashed by sds string, stores serverCommand structs. */
+hashsetType commandSetType = {.elementGetKey = hashsetCommandGetKey,
+                              .hashFunction = dictSdsCaseHash,
+                              .keyCompare = hashsetStringKeyCaseCompare,
+                              .instant_rehashing = 1};
+
+/* Command set, hashed by char* string, stores serverCommand structs. */
```

Comment:
```suggestion
/* Sub-command set, hashed by char* string, stores serverCommand structs. */
```
Is there a reason these need to be different? It seems like we could use declared name for both.

Reply:
@SoftlyRaining knows this better.

I suppose it's about renamed commands or something, where the declared name and the real name are not the same.

Reply:
Oh, I totally forgot about renamed commands here. @SoftlyRaining is that why it was done this way?

Reply:
I had to go figure it out again because I forgot, but I think I got it: Given a subcommand like "PARENTCMD SUBCMD ARG1 ARG2", declared_name would be "SUBCMD" (which could overlap with other subcommands) and fullname would be "PARENTCMD|SUBCMD". (see `populateCommandStructure()`)

I don't immediately see why `fullname` couldn't be used for both cases though

Reply:
Anyway, this logic is not new, right? It's just changed from dict to hashset?

Reply:
Yes, it is a 1:1 port, there should be no logic change as far as how commands work.